pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -138,11 +155,34 @@ func vmStatsCollectorCallback() []operatormetrics.CollectorResult {
 	}
 
 	var results []operatormetrics.CollectorResult
+	results = append(results, CollectVMsInfo(vms)...)
 	results = append(results, CollectResourceRequests(vms)...)
 	results = append(results, reportVmsStats(vms)...)
 	return results
 }
 
+func CollectVMsInfo(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var results []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		os, workload, flavor := getSystemInfoFromAnnotations(vm.Spec.Template.ObjectMeta.Annotations)
+		instanceType := getInstancetypeFromAnnotations(vm.Spec.Template.ObjectMeta.Annotations)
```

Comment:
I don't think the instancetype / preference annotation will be available at this point. The should only be present on the VMI. You can get then instancetype and preference from `spec.instancetype` and `spec.preference` instead.

Reply:
Great catch, thank you. Updated